pkg/app/web/src/components/applications-page/application-filter/index.tsx

```diff
@@ -55,9 +60,20 @@ export const ApplicationFilter: FC<ApplicationFilterProps> = memo(
     };
 
     const [labelOptions, setLabelOptions] = useState(new Array<string>());
-
     const [selectedLabels, setSelectedLabels] = useState(new Array<string>());
 
+    useEffect(() => {
+      const allLabels = new Array<string>();
+      applications
+        .filter((app) => app.labelsMap.length > 0)
+        .map((app) => {
+          app.labelsMap.map((label) => {
+            allLabels.push(`${label[0]}:${label[1]}`);
```

Comment:
Don't we have to get rid of duplicates?

Reply:
that's true 😄

Reply:
@nakabonne , instead of making labelsMap right here, using useEffect to update labelsMap whenever applications list fetched from the storage is updated may be better.

Reply:
You mean managing labels in the local storage and letting the application module to update it, right? Sounds nice!